src/core/thread/mle.cpp

```diff
@@ -506,9 +508,29 @@ otError Mle::Discover(const Mac::ChannelMask &aScanChannels,
     MeshCoP::DiscoveryRequestTlv discoveryRequest;
     uint16_t                     startOffset;
 
-    VerifyOrExit(!mIsDiscoverInProgress, error = OT_ERROR_BUSY);
+    VerifyOrExit(!mDiscoverInProgress, error = OT_ERROR_BUSY);
 
-    mEnableEui64Filtering = aEnableEui64Filtering;
+    mDiscoverEnableFiltering = aEnableFiltering;
```

Comment:
Is it better to implement this filtering in joiner(`Joiner::HandleDiscoverResult()`)? I think this keeps the interface clean and is more extensible.

Reply:
The EUI64 filtering was already added to MLE discover scan from https://github.com/openthread/openthread/pull/1685 along with OOB steering data control.  I am just updating the existing implementation in this PR (first commit).

Note that the MLE discover scan EUI64 filtering is already part of the public API `otThreadDiscover(.., aEnableEui64Filtering,...)` and exposed through spinel and wpantund. So we don't want to  remove it from MLE discover scan (IIRC, this feature was added (for internal use) to be able to filter scan result during weave pairing).

In this PR, we are actually removing the redundant filtering code in `Joiner` and use the existing one in MLE discover scan.